engine/src/main/java/io/github/trimax/venta/engine/utils/ArchiveUtil.java

```diff
@@ -32,7 +32,11 @@ public List<ResourceService.Resource> load(@NonNull final File archive) {
     @SneakyThrows
     private void load(@NonNull final File archive, @NonNull final TriConsumer<String, GroupType, byte[]> resourceLoader) {
         try (var in = new DataInputStream(new GZIPInputStream(new BufferedInputStream(new FileInputStream(archive))))) {
-            loadNode(in, resourceLoader, "");
+            while (in.available() > 0) {
+                final var groupName = in.readUTF();
```

Comment:
Let's move those 3 lines of while into a separate method: loadGroup, so that it will look like during writing

Reply:
And one more thing. Now we write group twice. We can do it just once, right? Well, we actually don't need to write this group name since we don't use it :D